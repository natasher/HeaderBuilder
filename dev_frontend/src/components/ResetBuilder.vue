<template>
  <a id            = "reset-builder"
    @click.prevent = "resetBuilder({ $modal: $modal })">
    Reset builder
  </a>
</template>

<script>
import { mapActions } from 'vuex'

export default {

  name: 'ResetBuilder',

  methods: mapActions( 'devices', [ 'resetBuilder' ]),

}
</script>
