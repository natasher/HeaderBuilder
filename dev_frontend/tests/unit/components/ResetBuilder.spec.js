import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex          from 'vuex'
import flushPromises from 'flush-promises'
import merge         from 'lodash.merge'
import ResetBuilder  from '../../../src/components/ResetBuilder.vue'

const localVue = createLocalVue()
localVue.use( Vuex )

function createStore(overrides) {
  const defaultStoreConfig = {
    modules: {
      devices: {
        namespaced: true,
        actions: {
          resetBuilder: jest.fn(),
        }
      }
    }
  }

  return new Vuex.Store(
    merge( defaultStoreConfig, overrides )
  )

}

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    mocks: {
      $modal: jest.fn()
    },
    store: createStore()
  }

  return shallowMount( ResetBuilder, merge( defaultMountingOptions, overrides ))
}

describe('ResetBuilder.vue', () => {

  test('should match snapshot', () => {
    const wrapper = createWrapper()

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should invoke `resetBuilder` action when clicked', async () => {
    const resetBuilder = jest.fn(() => Promise.resolve())
    const store = createStore({
      modules: {
        devices: {
          actions: {
            resetBuilder,
          }
        }
      }
    })
    const wrapper = createWrapper({ store, })

    wrapper.trigger('click')
    await flushPromises()

    expect( resetBuilder ).toBeCalled()
  })

})