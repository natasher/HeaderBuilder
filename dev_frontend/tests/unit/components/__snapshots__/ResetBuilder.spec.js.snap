// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetBuilder.vue should match snapshot 1`] = `
<a
  id="reset-builder"
>
  
  Reset builder

</a>
`;
